src/scratchtocatrobat/converter/converter.py

```diff
@@ -1858,7 +1858,9 @@ def _convert_variable_block(self):
         if user_variable is None:
             # WORKAROUND: for generated variables added in preprocessing step
             # must be generated user variable, otherwise the variable must have already been added at this stage!
-            assert(_is_generated(variable_name))
+            if not _is_generated(variable_name):
+                log.warning("Variable with name :'"+variable_name+"' does not exist. Creating it now.")
+                catformula.UserVariable(generated_variable_name(variable_name))
```

Comment:
Please use:

if user_variable is None and not _is_generated(variable_name)

To remove one level of indentation.

Reply:
by taking again a look into this I found out that the line ```catformula.UserVariable(generated_variable_name(variable_name))``` 
is redundant and not needed therfore I removed it with the new changes.

However I don't think we should make a single If for this since generating variables would fail if we do move
```
 catrobat.add_user_variable(self.project, variable_name, self.sprite, self.sprite.getName())
            user_variable = self.scene.getDataContainer().getUserVariable(self.sprite, variable_name)
```
inside the scope of if not_is_generated(variable_name)